Still Needs Work

clburns107 opened the pull request.

sumeetjain commented:
:+1:

Commit:
fin

Commit:
fin

Commit:
fin

sumeetjain commented on lib/reverse_string.rb:
```diff
@@ -1,5 +1,21 @@
 def reverse_characters(str)
+#Ok this is a little better but im still not sure how to use the .join to get it back to a string
+#The array can be called back backwards but its not "saved" anywhere in that order so it cant be converted to a string
+string="Today is a good day"
```
This method is meant to be generic–it should work for **any** string at all. And that hypothetical string is already represented by `str` (the "parameter" of this method). So you can remove `string="Today is a good day"`–just imagine you have a string already, and it's stored (already) in a variable named `str`.

sumeetjain commented on lib/reverse_string.rb:
```diff
@@ -1,5 +1,21 @@
+#Ok this is a little better but im still not sure how to use the .join to get it back to a string
+#The array can be called back backwards but its not "saved" anywhere in that order so it cant be converted to a string
+string="Today is a good day"
+array=string.split
```
You'll want to replace `array=string.split` with `array=str.split`.

sumeetjain commented on lib/reverse_string.rb:
```diff
@@ -1,5 +1,21 @@
+x=0
+for loop in array
+puts array[x-=1]
 end
```
This `end` is ending the `for` loop, but where did the `end` for the method itself go? I think you accidentally lost it in the coding process :)

sumeetjain commented on lib/reverse_string.rb:
```diff
@@ -1,5 +1,21 @@
+    puts array[x]   
+    
 end
+    
```
As with the above, you've lost an `end` :) This `end` ends the `array.length.times` loop, but the `end` that ends the method itself is gone!

sumeetjain commented on lib/reverse_string.rb:
```diff
@@ -1,5 +1,21 @@
 def reverse_characters(str)
+#Ok this is a little better but im still not sure how to use the .join to get it back to a string
+#The array can be called back backwards but its not "saved" anywhere in that order so it cant be converted to a string
```
Instead of using `puts` (which you don't need to use in this method at all), consider creating an empty array at the beginning of the method–and then push each character (starting from the end, of course) into that array. That way, you'll be saving the characters in their reverse order as you go through it.

Commit:
fin

sumeetjain commented:
Nice! Fix the indentation, and I'll approve after that.

clburns107 commented:
This is the message I'm getting right now when I try to submit my changes.  This happens a lot in Cloud9 and I'm not sure why. 

On branch master
Your branch is up-to-date with 'origin/master'.

nothing to commit, working directory clean
So here is the formatting correction:

def reverse_characters(str)

arr_fin=[]
array=str.split("")
x=0

for loop in array
  arr_fin<<array[x-=1]
end
arr_fin.join""
end

def reverse_words(str)

arr_fin=[]
array=str.split
x=0

for loop in array
  arr_fin<<array[x-=1]
end
arr_fin.join" "
end

sumeetjain commented:
Hm. That sounds like it would be caused by your not having saved the files. See if you can manually use the **File > Save** menu to save the files. Sometimes you have to manually click inside the code editing area first and then go to the **File** menu. Otherwise, the **Save** action might not be activated. It's weird.

sumeetjain commented:
I'm going to approve this anyway, but please do check on the saving issue.

clburns107 commented:
I ended up just clearing out all of the code, saving, and retyping it from scratch. Seems to rake and push just fine now.
